app/controllers/coronavirus_form/product_details_controller.rb

```diff
@@ -78,27 +81,61 @@ def validate_missing_fields
     required.each_with_object([]) do |field, invalid_fields|
       next if @product[field.to_sym].present?
 
-      invalid_fields << {
-        field: field,
-        text: t("coronavirus_form.questions.#{controller_name}.#{field}.custom_error",
-                default: t("coronavirus_form.errors.missing_mandatory_text_field",
-                           field: t("coronavirus_form.questions.#{controller_name}.#{field}.label")).humanize),
-      }
+      invalid_fields << error_response(field)
     end
   end
 
+  def validate_product_quantity
+    field = "product_quantity"
+    quantity = Integer(@product[field.to_sym])
+    return if quantity && quantity.positive?
+
+    error_response(field)
+  rescue TypeError, ArgumentError
+    error_response(field)
+  end
+
+  def validate_product_cost
+    field = "product_cost"
+    cost = Float(@product[field.to_sym])
+    return if cost && (cost.positive? || cost.zero?) && @product[field.to_sym].match?(/^\d+\.?\d{0,2}$/)
```

Comment:
Would it be possible to create a constant for the regex? E.g. `VALID_FLOAT = /^\d+\.?\d{0,2}$/`, I usually prefer this since one can see at a glance what the intent of the regex is.